File: mobile/src/main/java/com/alexstyl/specialdates/events/namedays/calendar/resource/SpecialNamedays.java
package com.alexstyl.specialdates.events.namedays.calendar.resource;

import com.alexstyl.specialdates.date.Date;
import com.alexstyl.specialdates.events.namedays.NameCelebrations;
import com.alexstyl.specialdates.events.namedays.NamesInADate;

import java.util.List;

public interface SpecialNamedays {

    NamesInADate getNamedayOn(Date date);

    NameCelebrations getNamedaysFor(String name, int year);

    List<String> getAllNames();
}

File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/calendar/resource/SpecialNamedays.kt
package com.alexstyl.specialdates.events.namedays.calendar.resource

import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.events.namedays.NameCelebrations
import com.alexstyl.specialdates.events.namedays.NamesInADate

interface SpecialNamedays {

    val allNames: List<String>

    fun getNamedayOn(date: Date): NamesInADate

    fun getNamedaysFor(name: String, year: Int): NameCelebrations
}
